src/front-end/graphics.js

```diff
@@ -2,483 +2,526 @@
+    // DEBUG: showing control points and poly line
+    if (DEBUG) {
+      var polyLine = s.polyline(Utils.allToScreen(properties.path));
+      var controlPts = _renderPoints(s, properties.path);
+      polyLine.attr({
+        fill: "transparent",
+        strokeWidth: 5,
+        stroke: Utils.hex(color[0], color[1], color[2]),
+        strokeDasharray: "10"
+      });
+      // return s.g(polyLine, controlPts, curve)
+      return s.g(polyLine, curve)
     }
+    return curve
+  }
+
+  /**
+  * Renders an arrow on the canvas
+  * Arrow can be either a regular or a curved arrow, specified by style
+  * @param       {Snap} s  Snap.svg global object
+  * @param       {JSON} obj JSON object from Haskell server
+  */
+  function _renderArrow(s, properties) {
+    var style    = properties.style
+    var [sx, sy] = Utils.scr([properties.startX, properties.startY])
+    var [ex, ey] = Utils.scr([properties.endX, properties.endY])
+    var thickness = properties.thickness / 6
+    var color = properties.color
+    var len = Snap.len(ex, ey, sx, sy)
+
+    var body_path = [0, thickness, len - 5 * thickness, thickness,
+      len - 5 * thickness, -1 * thickness, 0, -1 * thickness]
+      var head_path = [len - 5 * thickness, 3 * thickness, len, 0,
+        len - 5 * thickness, -3 * thickness]
 
-    /**
-     * Renders a point on the canvas
-     * @param       {Snap} s  Snap.svg global object
-     * @param       {JSON} obj JSON object from Haskell server
-     */
-     // FIXME: point radius is hardcoded
-     // FIXME: render colored dot
-    function _renderPoint(s, properties) {
-        [x, y] = [properties.x, properties.y]
-        var pt = s.circle(x, y, 4);
-        var color = properties.color
-        pt.attr({
-            fill: "#000000",
-            "fill-opacity": 1
-        });
-        return pt
-    }
+        var angle = Snap.angle(ex, ey, sx, sy)
+        var myMatrix = new Snap.Matrix();
+        myMatrix.translate(sx , sy);
+        myMatrix.rotate(angle, 0, 0);
 
-    /**
-     * Renders a square on the canvas
-     * @param       {Snap} s  Snap.svg global object
-     * @param       {JSON} obj JSON object from Haskell server
-     */
-    function _renderSquare(s, properties) {
-        [x, y] = [properties.x, properties.y]
-        var side = properties.side
-        var sq = s.rect(x - side/2, y - side/2, side, side);
-        var color = properties.color
-        sq.attr({
-            fill: Utils.hex(color[0], color[1], color[2]),
-            "fill-opacity": color[3],
-            "stroke-width": properties['stroke-width'],
-            "stroke": "black"
-        });
-        if(properties.stylec == "dashed") {
-            sq.attr({
-                strokeDasharray: "7, 5"
-            })
+        if(style == "straight"){
+          var line = s.polygon(body_path).transform(myMatrix.toTransformString())
+          line.attr({
+            "fill-opacity" : 1,
+            fill : Utils.hex(color[0], color[1], color[2]),
```

Comment:
fill-opacity should be set programmatically, right? using color[3] (the alpha attribute)